cmd/geth/chaincmd.go

```diff
@@ -207,13 +207,41 @@ func initGenesis(ctx *cli.Context) error {
 }
 
 func dumpGenesis(ctx *cli.Context) error {
-	// TODO(rjl493456442) support loading from the custom datadir
-	genesis := utils.MakeGenesis(ctx)
-	if genesis == nil {
-		genesis = core.DefaultGenesisBlock()
+	// if there is a testnet preset enabled, dump that
+	if utils.IsTestnetPreset(ctx) {
```

Comment:
Personally I think mainnet should also be handled here. If users want to dump out mainnet genesis, he has to specify `--mainnet`.